lib/Dialect/TritonGPU/IR/Types.cpp

```diff
@@ -167,6 +166,41 @@ LogicalResult MemDescType::verify(function_ref<InFlightDiagnostic()> emitError,
   } else {
     return emitError() << encoding << " is not a valid encoding";
   }
+
+  // PaddedSharedEncodingAttr is also a SharedEncodingTrait but we have some
+  // additional rules to verify.
+  if (auto enc = dyn_cast<PaddedSharedEncodingAttr>(encoding)) {
+    auto rank = enc.getRank();
+
+    if (rank != shape.size() && rank != shape.size() - 1) {
+      return emitError() << "padding rank must be equal to or one less than "
+                         << "the shape size when pipelining.";
+    }
+
+    // Subslices are not yet implemented
+    auto subsliceAllocSize =
+        allocShape.drop_front(allocShape.size() - shape.size());
+    for (auto [allocDim, shapeDim] : llvm::zip(shape, subsliceAllocSize)) {
+      if (allocDim != shapeDim) {
+        return emitError() << "Subslices with padded encodings are not yet "
+                           << "implemented.";
+      }
+    }
+
+    // Ensure linear component's outDims match the alloc size ignoring
+    // pipelining dimension
+    auto outDims = standardOutDimNames(ctx, rank);
+    auto ll = enc.getLinearComponent();
+    auto llAllocSize = allocShape.drop_front(allocShape.size() - rank);
```

Comment:
Just check against the shape, right?

Reply:
ugh, yeah, that's super annoying. We'll fix that in the near future tho by making the allocation dimension something different, rather than having  it being part of the shape.